### DISCUSSION TITLE
Poll: Can we drop Internet Explorer support in alphaTab

### DISCUSSION THREAD

**Discussion by Danielku15** [CoderLine/alphaTab#460] [General]
As everyone knows the Internet Explorer is slowly dying and all devs around the world are happy about that. But still it manages to keep a bit of market share and it's hard to judge if you can abandon it already. 

My questions to you all would be: Does anyone of you still need Internet Explorer support for alphaTab or would you be fine if we drop it? 

Vote by adding reactions to this post and feel free to add further comments 

👍  for dropping IE support
👎  for keeping it 

Pinging some known stakeholders of the web flavor: @cheap-glitch @MDoder97 @LeadGuitarMx @yaras6 @dgriessler @gburlet @chris-saunders

**Comment by thoun** [comment]
If IE support is dropped, will you plan usage of Intersection Observer to do lazy rendering ? I believe actual usage is based on spying window scroll and resize, which is OK on most cases but not all : tab appearing on JS dom modification, with no scroll/resize will wait next scroll/resize event to display.

https://caniuse.com/intersectionobserver

**Reply by Danielku15** [reply]
I do not have many direct plans after dropping IE support yet. So far on my list is: 

- Use again `dominant-baseline` for vertical alignment
- Remove of some old code (vba loader etc.)
- Start modernizing some APIs with promises and async stuff
- Compiling modern ES* code to have a smaller footprint. 

Using Intersection Observer sounds nice and would also give a performance boost and is worth some consideration.

**Comment by thoun** [comment]
When do you plan to end this poll ? It seems already lost for IE (yeah !)

**Reply by Danielku15** [reply]
I did not close it officially but with the votes we got I think I will drop IE support in version 1.2. I locally already started cleaning some things on a new branch.

**Comment by Danielku15** [comment]
@bbevren You voted to keep IE support, can you elaborate why? Can you still notice IE usage on your website?

**Comment by Danielku15** [comment]
Update: Follow the progress on the IE removal and related improvements here https://github.com/CoderLine/alphaTab/issues/507 Most of the things I plan to still do in 1.2

**Comment by gburlet** [comment]
My vote is to kill IE with fire 🔥
